tests/unit/testsuite/Bolt/Boltpay/Helper/GraphQLTraitTest.php

```diff
@@ -0,0 +1,100 @@
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->setMethods(array('getContextInfo', 'addMetaData', 'getBoltPluginVersion', 'getApiUrl', 'notifyException', 'logException'))
+            ->getMockForTrait();
+        $this->currentMock->method('getBoltPluginVersion')->willReturn('2.3.0');
+        $this->currentMock->method('getApiUrl')->willReturn('https://api.bolt.com/');
+
+
+        $this->guzzleClientMock = $this->getMockBuilder('Boltpay_Guzzle_ApiClient')
+            ->disableOriginalConstructor()
+            ->setMethods(array('post'))
+            ->getMock();
+
+        $this->responseMock = $this->getMockForAbstractClass(
+            '\Psr\Http\Message\ResponseInterface',
+            [],
+            '',
+            false,
+            true,
+            true,
+            ['getBody']
+        );
+
+        Bolt_Boltpay_TestHelper::setNonPublicProperty(
+            $this->currentMock,
+            'apiClient',
+            $this->guzzleClientMock
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function getFeatureSwitches_success()
+    {
+        $this->guzzleClientMock->expects($this->once())->method('post')->with(
+            'https://api.bolt.com/v2/merchant/api',
+            $this->callback(function ($jsonData) {
+                $testJsonData = '{"operationName":"GetFeatureSwitches","variables":{"type":"MAGENTO_1","version":"2.3.0"},"query":"query GetFeatureSwitches($type: PluginType!, $version: String!) {\n  plugin(type: $type, version: $version) {\n    features {\n      name\n      value\n      defaultValue\n      rolloutPercentage\n    }\n  }\n}"}';
+                return str_replace('\r\n', '\n', $jsonData) == $testJsonData;
+            }),
+            $this->anything()
+        )->willReturn($this->responseMock);
+
+        $this->currentMock->method('getContextInfo')->willReturn(array());
+
+        $this->responseMock->expects($this->once())->method('getBody')
+            ->willReturn('{"data": {"features": {"name": "OK", "value": true, "default_value": false, "rollout_percentage": 100}}}');
+
+        $response = $this->currentMock->getFeatureSwitches();
+        $this->assertEquals("OK", $response->data->features->name);
+    }
+
+    /**
+     * @test
+     *
+     * @expectedException Exception
+     * @expectedExceptionMessage Test exception
+     */
+    public function getFeatureSwitches_fail()
+    {
+        $exception = new Exception('Test exception');
+        $this->guzzleClientMock->expects($this->once())->method('post')->will($this->throwException($exception));
+
+        $this->currentMock->expects($this->once())->method('notifyException')->with($exception);
+        $this->currentMock->expects($this->once())->method('logException')->with($exception);
+
+        $response = $this->currentMock->getFeatureSwitches();
+        $this->assertEquals("OK", $response->data->features->name);
```

Comment:
@vitaliyreznikov 
This test and that it passes is curious to me.  There are a couple of things that concern me.  

First, considering that we are saying that an exception is thrown on attempting to make the GQL call, how is it that we are making an assertion about the return value?  If an exception was thrown, we shouldn't reach the point in the code where a value is returned, right?  It doesn't appear that way to me. 

Second, even if somehow we are suppose to get a return value, neither in this test method, nor in the `setUp` do I see the response being mocked.  It appears that we are relying on byproducts of the previous test.  This is not good form.  If this your intention, then you should consider moving the code for the response mock to the `setUp` method for better form.

However, it is my belief is that internally, PHPUnit is throwing an exception at line 96, 
```
$response = $this->currentMock->getFeatureSwitches();
```
catches it and then evaluates the exception annotations that are in the PHPDoc.  I do not believe that line 97
```
 $this->assertEquals("OK", $response->data->features->name);
```
is ever exectuted and absolutely does not make sense.  You can test my theory by changing line 97 to 
```
$this->assertEquals(0, 1);
```
and I believe that the test will still pass.

Reply:
Thank you Leon! I removed this line